web/yo/app/scripts/directives/pubiframe.js

```diff
@@ -22,16 +22,21 @@ angular.module('oncokbApp')
                     modelValue = S(modelValue).stripTags().s;
                     modelValue = S(modelValue).collapseWhitespace().s;
                     var pubs = FindRegex.result(modelValue);
-                    scope.pubs = pubs;
-                    // if (pubs.length === 0) {
-                    //     scope.pubs = [];
-                    // } else {
-                    //     FindRegex.validation(pubs).then(function(result) {
-                    //         scope.pubs = result;
-                    //     }, function (error) {
-                    //         console.log('Error happened', error);
-                    //     });
-                    // }
+                    if (pubs.length > 0) {
+                        var pubsCache = pubUtils.get(pubs);
+                        if (pubsCache.notValidatedPuds.length > 0) {
+                            FindRegex.validation(pubsCache.notValidatedPuds).then(function(result) {
+                                pubUtils.set(result);
+                                scope.pubs = _.union(pubsCache.validatedPuds, result);
```

Comment:
There would not be any duplicate in validatedPuds from result.

Reply:
Right. I'll use _.concat() in here.

Reply:
Why there are duplications?

Reply:
Just in case curators will type same PMID twice.